clap-v3-utils/src/input_parsers/signer.rs

```diff
@@ -15,6 +18,90 @@ use {
 // Sentinel value used to indicate to write to screen instead of file
 pub const STDOUT_OUTFILE_TOKEN: &str = "-";
 
+#[derive(Debug)]
+pub struct SignerSourceParserBuilder {
+    prompt: bool,
+    file_path: bool,
+    usb: bool,
+    stdin: bool,
+    pubkey: bool,
+    allow_legacy: bool,
+}
+impl Default for SignerSourceParserBuilder {
+    fn default() -> Self {
+        Self {
+            prompt: true,
+            file_path: true,
+            usb: true,
+            stdin: true,
+            pubkey: true,
+            allow_legacy: true,
+        }
+    }
+}
+
+impl SignerSourceParserBuilder {
+    pub fn new() -> Self {
```

Comment:
usually `T::new(...)` is like `T::default()` with extra steps.  it's a little confusing that they're exactly the opposite here with no indication in the names.  i'd suggest dropping one of them

Reply:
Valid point. I removed `new` and ended up deriving the `Default` trait. I personally feel like `new` is more natural, but it seems like clippy favors default. Let me know what you think.